File: lists/sum.py


def for_summer(l):
    s = 0
    for i in l:
        s += i 

    return s 

def while_sum(l):
    s = 0
    x = 0
    while x < len(l):
        s += l[x]
        x += 1
    return s 

def recur_sum(l):
    return reduce((lambda a, b: a + b), l)


temp = [3, 5, 6, 7, 8, 9]

print(for_summer(temp))
print(while_sum(temp))
print(recur_sum(temp))

File: lists/sum.js


const for_sum = (arr) => {
    let total = 0;
    for (a in arr) {
        total += arr[a];
    }

    return total;
}

const while_sum = arr => {
    let total = 0;
    while (arr.length > 0) {
        total += +arr.pop();
    }

    return total;
}

const recur_sum = arr => {
    let num = arr.reduce((a, b) => a + b);
    return num;
}

let temp = [2, 4, 5, 6, 7, 8, 9];

console.log(for_sum(temp));

console.log(recur_sum(temp));
console.log(while_sum(temp));